src/pkg/egress/syslog/filtering_drain_writer.go

```diff
@@ -33,26 +35,54 @@ func NewFilteringDrainWriter(binding Binding, writer egress.Writer) (*FilteringD
 }
 
 func (w *FilteringDrainWriter) Write(env *loggregator_v2.Envelope) error {
-	if w.binding.Type == BINDING_TYPE_AGGREGATE {
+	if w.binding.DrainData == ALL {
 		return w.writer.Write(env)
 	}
 
-	if env.GetTimer() != nil || env.GetEvent() != nil {
-		return nil
+	if env.GetTimer() != nil {
```

Comment:
Is it possible for an envelope to respond to multiple types? e.g. a scenario where `env.GetTimer()` and `env.GetEvent()` are both non-nil

Reply:
I don't believe so.